drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -653,6 +674,7 @@ static int i2c_mspm0g3xxx_init(const struct device *dev)
 	int ret;
 
 	k_sem_init(&data->i2c_busy_sem, 0, 1);
+	k_sem_init(&data->transfer_timeout_sem, 1, 1);
```

Comment:
> So as I understand this, basically you grab take the semaphore as it is created as well. Are we guaranteed that one of > the places that release it are called prior to the first transfer?

If you look closely we're actually *never* waiting for the semaphore *unless* it's after we've already loaded it and triggered a transfer. Ie. if the lock is already locked (if that was the case) the initial lock will "fail" but it doesn't matter, as we only care about the lock being locked so we can wait for it to be unlocked :) Therefore I'm also not unlocking it after locking it after waiting - simply because it doesn't make a difference. Performance-wise it's faster and code-wise it would look like it's critical to do so. If that makes sense :)

> I'm thinking that there might be missing a give at the end of this function

Good spotted :wink: The syntax is actually the opposite; so by giving a count of 1 it means that 1 more lock can be performed. So this semaphore is in fact not locked during init.

Reply:
Got it! Thanks :) Lol, that API got me good haha